### ISSUE TITLE
Steins;Gate 0 (825630)

### ISSUE THREAD

**Issue by RCal-Dev** [ValveSoftware/Proton#216]
Decided to try a Visual Novel and see if it would run. Steins;gate worked flawlessly, Steins;gate 0 however fails to start. 
[steam-825630.log](https://github.com/ValveSoftware/Proton/files/2312807/steam-825630.log)

**Comment by Vladimir-A** [comment]
The game has bat file. Need "start boot.bat" or "start launcher.exe EN". In wine cmd it was easy.

**Comment by RCal-Dev** [comment]
Alright, thank you, let me try that.

**Comment by RCal-Dev** [comment]
Confirmed, I can get it to start using wine cmd (with Proton). I'd still like to be able to launch it through steam itself, however, so I'd like to keep the issue up.

To be clear, it does not run when I click "Play" on the title in the Steam Library, however, if I use proton's wine with CMD, the games (launcher) launches fine.

**Comment by ghost** [comment]
I'm able to run the launcher dialog, but when I click the "Start game" button it crash to this error.

"wine: Unhandled page fault on read access to 0x00000000 at address 0x4a7f61 (thread 01be), starting debugger..."

I run the launcher from boot.bat with the proton python file like this:

First:
On a bash, export the COMPAT_DATA variable:

export STEAM_COMPAT_DATA_PATH="/media/walkuren/WD 4TB 4/SteamLibrary/steamapps/compatdata/825630"

Open a explorer window from proton command:

./proton run "/media/walkuren/WD 4TB 4/SteamLibrary/steamapps/compatdata/pfx/dosdev
ices/z:/media/walkuren/WD 4TB 4/SteamLibrary/steamapps/common/STEINS;GATE 0/"

Second:
In the explorer window that opens from the above command I go to "C:/windows/system32/" and open cmd.exe, then go to the game folder with the command "cd /d Z:\media\walkuren\WD 4TB 4\SteamLibrary\steamapps\common\STEINS;GATE 0\" 

Third:
I open the launcher just with the boot.bat file from cmd, it opens but crash when attempt to play
I also try with this commands:

![captura de pantalla de 2018-08-25 02-20-58](https://user-images.githubusercontent.com/42692685/44616888-b20c6c80-a80d-11e8-9a4a-f82dee42e7bf.png)


start Game.exe EN
start launcher.exe EN

All of them with the same result.

"wine: Unhandled page fault on read access to 0x00000000 at address 0x4a7f61 (thread 01be), starting debugger..."


I've noted that the steam app window go to playing status with my user name colored green and the game status showing the "Running..." status

also

The a console that opens for a second from the play button on the steam app show the message:
"boot.bat is not recognized as an internal or external command, operable program or batch file."

if you dont use the export STEAM_COMPAT_DATA_PATH and try to run the game it shows this message box:

![captura de pantalla de 2018-08-25 00-31-36](https://user-images.githubusercontent.com/42692685/44616833-65746180-a80c-11e8-8a33-cb31a2ef3788.png)


I suspect that the real problem to this, is that the steam app hardcoded the app launcher.dat to the boot.bat instead of the launcher.exe causing the game not to run from the steam app with proton
and this maybe is because the *.bat files not associated to cmd.exe on wine by default.


Well that was my attempt to run the game and a really interesting night with proton.

This was the only game that I couldn't run from my tested windows games on linux

I hope this helps people on trying to run steins;gate 0 on linux if you are able to run flawlessly please share the used method.

I'm using linux mint 18.3 cinnamon on a amd ryzen 2600x/gtx1080

**Comment by RCal-Dev** [comment]
Yeah, just tried to launch the game with the launcher using WINE and game crashed with the same page fault exception.

Interesting.

**Comment by ghost** [comment]
Someone added to the steamplay compatibility list the method to run the game from the play button.


rename:
1. boot.bat to boot.bat.bak
2. launcher.exe to boot.bat

Still i think it should be an official way to do this, at least better than modify yourself the game files.

**Comment by RCal-Dev** [comment]
Update: As of Proton 3.7-6 Beta, Steins;gate 0 now works perfectly if launcher.exe is renamed to boot.bat. Was able to start up game, sound played. Didn't go too far into the story.

**Comment by user9957** [comment]
If you don't want to mess with the game's files, you can put the following config in `user_settings.py`:

```
import os, sys

user_settings = dict()

game_id = None
if game_id == None and 'SteamAppId' in os.environ:
    game_id = int(os.environ["SteamAppId"])
if game_id == None and 'SteamGameId' in os.environ:
    game_id = int(os.environ["SteamGameId"])
if game_id == None and 'STEAM_COMPAT_DATA_PATH' in os.environ:
    game_id = int(os.path.basename(os.environ['STEAM_COMPAT_DATA_PATH']))

# STEINS;GATE 0
if game_id == 825630:
    sys.argv[2:] = [os.path.expanduser("~/.steam/steam/steamapps/common/STEINS;GATE 0/launcher.exe"), "EN"]
```

This also allows to tweak other games easily (`elif game_id == ...`).

**Comment by Encritary** [comment]
Worked for me after adding ``EN`` to launch options. Otherwise, it tells me to use "boot.bat"

**Comment by Vladimir-A** [comment]
[STEINS;GATE 0](https://steamdb.info/app/825630/) works with proton-4.11-13, but doesn't launch with proton-5.0-5b and proton-5.0-6-rc3.
For proton-4.11-13 need add "EN" to launch option and rename "launcher.exe" to "boot.bat".
[STEINS;GATE ELITE](https://steamdb.info/app/819030/) has same "boot.bat" file, but doesn't need any tricks and works fine with proton-5.0-6-rc3.
[Gist](https://gist.github.com/Vladimir-A/3887bb79ec955d66bebaeec41cd4d8c3)
Logs:
[steam-825630-proton-4.11-13.zip](https://github.com/ValveSoftware/Proton/files/4466936/steam-825630-proton-4.11-13.zip)
[steam-825630-proton-5.0-5b.zip](https://github.com/ValveSoftware/Proton/files/4466937/steam-825630-proton-5.0-5b.zip)
[steam-825630-proton-5.0-6-rc3-no-tricks.zip](https://github.com/ValveSoftware/Proton/files/4466938/steam-825630-proton-5.0-6-rc3-no-tricks.zip)
[steam-825630-proton-5.0-6-rc3.zip](https://github.com/ValveSoftware/Proton/files/4466939/steam-825630-proton-5.0-6-rc3.zip)

**Comment by kisak-valve** [comment]
@Vladimir-A, maybe try completely closing steam, then run `steam-runtime` from a terminal and check if that has an effect?

Separately, does setting the game's launch options to `PROTON_USE_WINED3D=1 PROTON_FORCE_LARGE_ADDRESS_AWARE=0 %command% EN` have an effect?

**Comment by Vladimir-A** [comment]
@kisak-valve, I enabled `steam-runtime` [Gist](https://gist.github.com/Vladimir-A/d54b46056833d5feea658b223f8ac0ca).
I tested without `PROTON_USE_WINED3D=1 PROTON_FORCE_LARGE_ADDRESS_AWARE=0`, with this options together and separately. No any effect.

Logs:
[steam-825630.zip](https://github.com/ValveSoftware/Proton/files/4469843/steam-825630.zip)

**Comment by TraceStar** [comment]
> @Vladimir-A, maybe try completely closing steam, then run `steam-runtime` from a terminal and check if that has an effect?
> 
> Separately, does setting the game's launch options to `PROTON_USE_WINED3D=1 PROTON_FORCE_LARGE_ADDRESS_AWARE=0 %command% EN` have an effect?

I could get up to the launcher by renaming launcher.exe to boot.bat but it goes to a blank screen when clicking start game and then crash. Running steam-runtime seems to have fixed this. I can now start the game.

Only Proton 3.16-9 works for me though. All other proton version fails to get even the launcher running.

No logfiles generated for  proton 4.2.9, and 4.11-13 and 5.0-7  when launcher fails to load. I didn't have proton logs enabled before running steam-runtime so can't upload the failed start game for 3.16-9

Running Solus 4.1 with Linux kernel 5.6.13 and Mesa 19.3.4. CPU Intel i7-4790 with Radeon RX480.
